### PR TITLE
Feat/provenance project node

### PR DESCRIPTION
**Pull request by marshHawk4**

### CODE REVIEW

File: askem.dbml
Comment (fivegrant) [inline askem.dbml:29]: Should this be in the `askem.dbml`? Isn't this part of the implementation?

Comment (fivegrant) [reply askem.dbml:29]: Couldn't it just be included here: https://github.com/DARPA-ASKEM/data-service/blob/2880f968cb64bb0809faf8b52a12e646227fcd4e/tds/schema/provenance.py#L21-L35

File: tds/routers/models.py
Comment (fivegrant) [inline tds/routers/models.py:402]:
```suggestion
            raise HTTPException(status_code=400, detail="Operation not supported")
```

File: askem.dbml
Comment (fivegrant) [inline askem.dbml:43]: Do we want this in the dbml?

File: scripts/provenance_fake_data.py
Comment (fivegrant) [inline scripts/provenance_fake_data.py:368]: Universal exception. How are you expecting things here to fail? Are you expecting any of the lines to fail or one in particular? Could we be using `try..except..else` or `try...except...finally`?

File: tds/db/graph/query_helpers.py
Comment (fivegrant) [inline tds/db/graph/query_helpers.py:22]:
```suggestion
        status_code=400, detail="Relationship direction is not allowed."
```

File: tds/db/graph/query_helpers.py
Comment (fivegrant) [inline tds/db/graph/query_helpers.py:50]:
```suggestion
        status_code=400, detail=f"Models can not be derived from this type: {root_type}"
```

File: tds/db/graph/search_provenance.py
Comment (fivegrant) [inline tds/db/graph/search_provenance.py:94]:
```suggestion
                status_code=400,
```

File: tds/db/graph/search_provenance.py
Comment (fivegrant) [inline tds/db/graph/search_provenance.py:124]:
```suggestion
                status_code=400,
```

File: tds/db/graph/search_provenance.py
Comment (fivegrant) [inline tds/db/graph/search_provenance.py]:
```suggestion
```

File: tds/db/graph/search_provenance.py
Comment (fivegrant) [inline tds/db/graph/search_provenance.py]:
```suggestion
            logger.info(query)
```

File: tds/db/graph/search_provenance.py
Comment (fivegrant) [inline tds/db/graph/search_provenance.py:163]:
```suggestion
```

File: tds/routers/models.py
Comment (fivegrant) [inline tds/routers/models.py:387]:
```suggestion
```

File: tds/routers/models.py
Comment (fivegrant) [inline tds/routers/models.py:537]:
```suggestion
```

File: tds/routers/models.py
Comment (fivegrant) [inline tds/routers/models.py:234]:
```suggestion
```

File: tds/routers/models.py
Comment (fivegrant) [inline tds/routers/models.py:246]:
```suggestion
```

File: tds/routers/models.py
Comment (fivegrant) [inline tds/routers/models.py:249]:
```suggestion
```

File: tds/routers/models.py
Comment (fivegrant) [inline tds/routers/models.py:326]:
```suggestion
        logger.info("Neo4j is set")
```

File: tds/db/graph/search_provenance.py
Comment (fivegrant) [inline tds/db/graph/search_provenance.py:59]:
```suggestion
```

File: tds/db/graph/search_provenance.py
Comment (fivegrant) [inline tds/db/graph/search_provenance.py:65]: What is `res`? Is it okay if this line is broken up. Just not sure what it's doing.

**Review by fivegrant — CHANGES_REQUESTED** [review]

**Comment by fivegrant** [comment]
rebase is needed because of this: https://github.com/DARPA-ASKEM/data-service/pull/143#pullrequestreview-1239324109

File: scripts/provenance_fake_data.py
Comment (marshHawk4) [reply scripts/provenance_fake_data.py]: removed

File: askem.dbml
Comment (marshHawk4) [reply askem.dbml:30]: I moved it out. Doesn't need to be there

File: tds/routers/models.py
Comment (marshHawk4) [reply tds/routers/models.py]: Changed

File: tds/routers/models.py
Comment (marshHawk4) [reply tds/routers/models.py]: removed

File: tds/db/graph/search_provenance.py
Comment (marshHawk4) [reply tds/db/graph/search_provenance.py:64]: res is just part of the result that is looping. It is going to be changed anyways in my next pr to account for returning the whole graph.

File: tds/routers/models.py
Comment (marshHawk4) [reply tds/routers/models.py]: updated

File: tds/db/graph/query_helpers.py
Comment (marshHawk4) [reply tds/db/graph/query_helpers.py]: updated

File: tds/settings.py
Comment (fivegrant) [inline tds/settings.py:22]: Do we want to this to be False by default? `NEO4J_ENABLED` could be set to `True` in `api.env.sample` so we wouldn't have to change it here.

File: tds/db/graph/search_provenance.py
Comment (fivegrant) [inline tds/db/graph/search_provenance.py:101]:
```suggestion
```

File: tds/db/graph/query_helpers.py
Comment (fivegrant) [inline tds/db/graph/query_helpers.py:46]:
```suggestion
            f"""
            Match(Pu:Publication {{id:{root_id}}})
            <-[r:EXTRACTED_FROM]-(In:Intermediate)
            Optional Match (In)<-[r2:REINTERPRETS *1..]-(Mr:ModelRevision)
            Optional Match(Mr)
            -[r3:EDITED_FROM|COPIED_FROM|GLUED_FROM|STRATIFED_FROM  *1..]-
            (Mr2:ModelRevision)
            with *,collect(r)+collect(r2)+collect(r3) as r4,
            collect(Mr)+collect(Mr2) as ms
            unwind ms as mss
            unwind r4 as r5
            Optional Match(mss)<-[r6:BEGINS_AT]-(Md:Model)
            with *, collect(r5)+collect(r6) as r7
            unwind r7 as r8 
            return Pu, In, ms,Md, r8
            """
```

File: tds/db/graph/query_helpers.py
Comment (fivegrant) [inline tds/db/graph/query_helpers.py:62]:
```suggestion
        return f"""
            Match (In:Intermediate {{id:{root_id}}})<-[r2:REINTERPRETS *1..]
            -(Mr:ModelRevision)
            Optional Match(Mr)
            -[r3:EDITED_FROM|COPIED_FROM|GLUED_FROM|STRATIFED_FROM  *1..]-
            (Mr2:ModelRevision)
            with *,collect(r2)+collect(r3) as r4, collect(Mr)+collect(Mr2) as ms
            unwind ms as mss
            unwind r4 as r5
            Optional Match(mss)<-[r6:BEGINS_AT]-(Md:Model)
            with *, collect(r5)+collect(r6) as r7
            unwind r7 as r8
            return  In, ms,Md, r8
            """
```

**Review by fivegrant — CHANGES_REQUESTED** [review]

File: tds/db/graph/search_provenance.py
Comment (marshHawk4) [reply tds/db/graph/search_provenance.py:101]: removed

File: tds/db/graph/query_helpers.py
Comment (marshHawk4) [reply tds/db/graph/query_helpers.py:62]: changed

File: tds/db/graph/search_provenance.py
Comment (fivegrant) [reply tds/db/graph/search_provenance.py:147]: git is wrong

**Review by fivegrant — APPROVED** [review]
Works for me, but we should wait until @mattprintz to merge

**Review by mattprintz — APPROVED** [review]
Looks good pending testing/sign-off on the startup script changes I made.

**Pull request merged**
